pipelines/api/utils.py

```diff
@@ -78,8 +78,11 @@ def _slugify_file(filename):
     return basename.rsplit('.', 1)[0]
 
 
-def _run_id_iterator(slug):
-    for sub_folder in os.listdir(slug):
+def _run_id_iterator(slug, log_limits):
+    dir_list = os.listdir(slug)
+    #sorted by modify time
+    dir_list = sorted(dir_list,key=lambda x: os.path.getmtime(os.path.join(slug, x)), reverse = True)
+    for sub_folder in dir_list[:log_limits]:
         if _is_valid_uuid(sub_folder):
```

Comment:
we need more error check for this function, e.g. https://github.com/Wiredcraft/pipelines/pull/134/files#diff-0521c268d7a59736de1c47ad52d4ab6e0ad52168e0aed7dae4b9201ae5504349R83-R98